packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -53,7 +53,7 @@ export class Repository extends RepositoryRef {
         const resource = new cloudformation.RepositoryResource(this, 'Resource', {
             repositoryName: props.repositoryName,
             // It says "Text", but they actually mean "Object".
-            repositoryPolicyText: this.policyDocument,
+            repositoryPolicyText: new cdk.Token(() => this.policyDocument),
```

Comment:
PolicyDocument should be a Token already, so this should not be necessary. Will investigate tomorrow.

Reply:
Actually, the test I added didn't pass without this change.

Reply:
I believe that. Will check into why tomorrow. Boo on me for not writing this test in the first place ;)

Reply:
Oh duh. It's not an object at construction time yet. It's lazily instantiated when you first call `addToResourcePolicy`.